handlers/register.go

```diff
@@ -0,0 +1,55 @@
+package handlers
+
+import (
+	"log"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/speedrun-website/leaderboard-backend/database"
+	"github.com/speedrun-website/leaderboard-backend/graph/model"
+	"github.com/speedrun-website/leaderboard-backend/utils"
+)
+
+func RegisterHandler(c *gin.Context) {
+	var registerValue model.Register
+
+	if err := c.Bind(&registerValue); err != nil {
+		log.Println("Unable to bind value", err)
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+			"message": err.Error(),
+		})
+		return
+	}
+
+	db, err := database.GetDatabase()
+
+	if err != nil {
+		log.Println("Unable to connect to database", err)
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+			"message": err.Error(),
+		})
+		return
+	}
+
+	var alreadyExist model.User
+	result := db.Where(model.User{Email: registerValue.Email}).Find(&alreadyExist)
+
+	if result.RowsAffected != 0 {
+		c.JSON(http.StatusBadRequest, gin.H{
```

Comment:
I don't think this should be a 400. It's not technically an obvious client error; usually I'd reserve 400 for truly malformed requests. There's a lot of different opinions on how to handle it instead, this Stack Overflow thread outlines a few of them. https://stackoverflow.com/a/53144807/2816588

Reply:
Interesting. I will switch to a 409 for now but also leaving a comment about potential issues.